staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -540,6 +543,53 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 			return err
 		}
 
+		// Check if the returned object needs to have its
+		// kubectl-client-side-apply managed fields migrated server-side-apply.
+		//
+		// By checking afterward instead of fetching the object beforehand and
+		// unconditionally fetching we can make 3 network requests in the rare
+		// case of migration and 1 request if migration is unnecessary.
+		//
+		// To check beforehand means 2 requests for most operations, and 3
+		//	requests in worst case.
+		if patchData, err := o.calculateSSAMigrationPatch(obj); err != nil {
+			// Print-error as a warning? Not sure how to treat this non-fatal error
+			// (non-fatal because object was successfully applied, but it might
+			// have issues since migration failed?)
+		} else if patchData != nil {
+			// If patchData is non-nil there is no patch necessary
```

Comment:
Not sure I understand this sentence. I would have written

`// Update object's managed fields with CSA managed fields merged into SSA manager.`

Reply:
Good catch, I forgot to comment, this one was definitely wrong, thanks Sean :-)